### ISSUE TITLE
Deprecation flag

### ISSUE THREAD

**Issue by davidcok** [dafny-lang/dafny#4037]
### Summary

Implement a new cli flag for /deprecation and remove the deprecation style option

### Background and Motivation

The old CLI has a /deprecation option . By default the new CLI warns about deprecated features, with no ability to turn them off. Also there is a value in the old CLI that enables warnings about '"deprecation style", of which the only example is unnecessary trailing semicolons (e.g. at the end of specification clauses).

### Proposed Feature

I suggest
1) either 
(a) creating a --warn-deprecation new-cli option, or
(b) deciding that dafny (in the new cli) will always warn about deprecated features without the ability to disable these warnings

2) converting the deprecated-style of extraneous semicolons into a normal deprecation warning. This will have the effect of creating warnings where users did not see them before if they have extraneous semicolons. It will also require us to deal with the quite a few tests that have extraneous semicolons.

### Alternatives

Status quo, which in the new CLI means
- deprecated features are always warned about
- extraneous semicolons are never warned about

**Comment by RustanLeino** [comment]
For 1, I'd be fine with either (a) or (b). My first reaction was that you'd expect a tool to allow you to turn off warnings, including deprecation warnings. But there's wisdom is always generating those warnings, since they're saying the program will not continue to be valid in the future.

For 2, I looked at where we're generating these warnings and it's only for `OldSemi`. I agree it's time to convert these to regular deprecation warnings.

Additional note about these semi-colons. Today, the parentheses in

``` dafny
ensures (MyLemma(); post)
```

are required. They would no longer be required if we got rid of `OldSemi` altogether. It occurs to me that we could also implement a different version of semi-colon parsing (in particular, parsing of `OldSemi` and parsing of the semi-colon that follows a lemma call in a `StmtExpr`). The different version could do a little lookahead to, in some case, be able to determine that a semi-colon must be an `OldSemi` rather than a lemma-`StmtExpr`. For example, if the token that follows the semi-colon is `ensures`, then the semi-colon is an `OldSemi`, but if the next token is `var` or `{`, then we don't know. The behavior of this different semi-colon parsing would be to give a warning and ignore the semi-colon if it is definite an `OldSemi`, and to parse it as a lemma-`StmtExpr` semi-colon otherwise. (This is different from the current behavior, in part because the current behavior favors the `OldSemi`.)
